In [1]:
import pandas as pd
import numpy as np
import math as math
from scipy import stats
from matplotlib import pyplot as plt

### Задача 1
О непрерывной равномерно распределенной случайной величине `B` известно, что ее дисперсия равна 0.2. Можно ли найти правую границу величины `B` и ее среднее значение зная, что левая граница равна 0.5? Если да, найдите ее.

#### Решение:
$$M(X) = \dfrac{a + b}{2}, \text{ } D(X) = \dfrac{(b - a)^2}{12},$$
где a - правая граница, b - левая граница.

b = 0.5, D(X) = 0.2.

Требуется определить a, M(X).

$\dfrac{(b - a)^2}{12} = 0.2 \text{ } => (0.5 - a)^2 = 12 * 0.2 \text{ } => (0.5 - a) = \sqrt {2.4} \text{ } => a = 0.5 - 2*\sqrt {0.6} = -1.05$

$M(X) = \dfrac{a + b}{2} = \dfrac{-1.05 + 0.5}{2} = -0.275$

#### Ответ:
a = -1.05

M(X) = -0.275

### Задача 2
Рост взрослого населения города `X` имеет нормальное распределение, причем средний рост равен 174 см, а среднее квадратическое отклонение равно 8 см. Посчитайте, какова вероятность того, что случайным образом выбранный взрослый человек имеет рост:
1. больше 182 см?
2. больше 190 см?
3. от 166 см до 190 см?
4. от 166 см до 182 см?
5. от 158 см до 190 см?
6. не выше 150 см или не ниже 190 см?
7. не выше 150 см или не ниже 198 см?
8. ниже 166 см?

Задачу можно решить двумя способами: без использования сторонних библиотек (`numpy`, `scipy`, `pandas` и пр.), а затем проверить себя с помощью встроенных функций

#### Решение:
$a = 174, \sigma = 8$

In [2]:
loc = 174 # mu
scale = 8 #sigma

In [3]:
# 1. рост больше 182 см?
x = 182
1 - stats.norm.cdf(x, loc, scale)

0.15865525393145707

In [4]:
# 2. рост больше 190 см?
x = 190
1 - stats.norm.cdf(x, loc, scale)

0.02275013194817921

In [5]:
# 3. рост от 160 см до 190 см?
x1 = 190
x2 = 160
stats.norm.cdf(x1, loc, scale) - stats.norm.cdf(x2, loc, scale)

0.9371907111880037

In [6]:
# 4. рост от 166 см до 182 см?
x1 = 182
x2 = 166
stats.norm.cdf(x1, loc, scale) - stats.norm.cdf(x2, loc, scale)

0.6826894921370859

In [7]:
# 5. рост от 158 см до 190 см?
x1 = 190
x2 = 158
stats.norm.cdf(x1, loc, scale) - stats.norm.cdf(x2, loc, scale)

0.9544997361036416

In [8]:
# 6. рост не выше 150 см или не ниже 190 см?
x1 = 190
x2 = 150
1 - stats.norm.cdf(x1, loc, scale) + stats.norm.cdf(x2, loc, scale)

0.0241000299798093

In [9]:
# 7. рост не выше 150 см или не ниже 198 см?
x1 = 198
x2 = 150
1 - stats.norm.cdf(x1, loc, scale) + stats.norm.cdf(x2, loc, scale)

0.0026997960632601965

In [10]:
# 8. рост ниже 166 см?
x = 166
stats.norm.cdf(x, loc, scale)

0.15865525393145707

#### Ответ:
1. 15.9%
2. 2.3%
3. 93.7%
4. 68.3%
5. 95.4%
6. 2.4%
7. 0.3%
8. 15.9%

### Задача 3
Коробки с шоколадом упаковываются автоматически. Их средняя масса равна 1.06 кг. Известно, что 5% коробок имеют массу, меньшую 1 кг. Найдите: а) среднее квадратическое отклонение массы коробки, б) процент коробок, имеющих массу больше 1.1 кг.

_Подсказка_. 1. Можно считать, что распределение массы коробки нормальное. 2. Найдите такое значение `scale`, для которого значение `cdf(x=1, loc=1.06, scale=scale)` близко к 0.05. Точности 0.0001 будет достаточно.

#### Решение:


In [11]:
# а) Найти среднее квадратичное отклонение с точностью 0.0001
scale = 0.0364
while abs(0.05 - stats.norm.cdf(x=1, loc=1.06, scale=scale)) > 0.0001:
    scale += 0.00001
print (scale)

0.03646000000000002


In [12]:
# б) процент коробок, имеющих массу больше 1.1 кг
1 - stats.norm.cdf(x=1.1, loc=1.06, scale=0.03646)

0.13630043457442875

#### Ответ:
а) 0.03646
б) 13.63%